### PR TITLE
Remove warning on component unmounted

### PR DESCRIPTION
**Pull request by kaelhem**
Hi,
Firstly, thanks for this library, it is timely for my project!

I made a simple update on the useEffect hook that loads Howler, in order to cancel actions after loading if the component is unmounted (to suppress warnings in console).
I also remove the use-on-mount file that seems useless.

Do not hesitate to tell me if you have questions or remarks.

Cheers

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -33,20 +31,32 @@ export default function useSound(
...
-        src: [url],
-        volume,
-        onload: handleLoad,
-        ...delegated,
-      });
-
-      setSound(sound);
-    });
-  });
+  useEffect(() => {
```
Comment (joshwcomeau) [inline src/index.ts:34]: Can't this still be done using the `useOnMount` helper? It should work exactly the same way

**Comment by kaelhem** [comment]
Sounds fair enough!
I've made the changes you wanted. Now, just the minimal code that fix the issue is added.

File: src/index.ts [hunk 2]
```diff
@@ -34,19 +34,27 @@ export default function useSound(
...
+            ...delegated,
+          });
+
+          setSound(sound);
+        }
       });
-
-      setSound(sound);
-    });
+    })();
```
Comment (joshwcomeau) [inline src/index.ts:54]: Hey @kaelhem, thanks for making the changes, and sorry for the delay.

I'm a bit confused by the IIFE 🤔 

I'm gonna merge your PR but remove it before publishing a new package. If this does have a benefit I'm not aware of, please let me know and I'll re-add it =)

**Comment by kaelhem** [comment]
"I'm a bit confused by the IIFE" yes me too 😃!
You were right to remove it.

**Pull request merged**